xarray/core/coordinates.py

```diff
@@ -208,13 +215,41 @@ def _to_dataset(self, shallow_copy=True):
     def to_dataset(self):
         return self._to_dataset()
 
+    def __setitem__(self, key, value):
```

Comment:
Just define this method once on `AbstractCoordinates` instead of repeating it twice